BootloaderCorePkg/Library/MpInitLib/Ia32/MpInitLib.c

```diff
@@ -96,6 +97,15 @@ CpuInit (
     SmmRebase (Index, ApicId, 0);
   }
 
+  PlatformCpuInitHook = (PLATFORM_CPU_INIT_HOOK)(UINTN)PcdGet32 (PcdPlatformCpuInitHook);
+  if (PlatformCpuInitHook != NULL) {
+    while (!AcquireSpinLockOrFail (&mMpDataStruct.SpinLock)) {
```

Comment:
I already commented in the old review.   Lock will slow down the MpInit.   We should only do it when it is required.   If lock is really required,  PlatformCpuInitHook function can add lock as required.  Common code can remove the spin lock.

Reply:
will remove it.